### PR TITLE
Feature/corpus document model

### LINKED ISSUE #1010: License / copyright information per corpus
In the interest of FAIR data, it would be nice if users could see what they are allowed to with downloaded csv data. I don't think any of our corpora specify this in their documentation at the moment.

We could add such information in the documentation file for the corpus. If we want to use a full license, it would make more sense to make that a separate page.

### LINKED ISSUE #1524: Change data model for corpus attachments
Currently, corpus definitions used "attached files", namely the image and various documentation files. The model is that the corpus definition specifies the filename, and this is resolved by looking in the directory that contains the corpus definition.

This system won't work for database-only corpora, so it should be adjusted.

### Corpus images

Use an `ImageField` to save the image file. The ImageField is a specialised version of `FileField`. See the [Django documentation on file management](https://docs.djangoproject.com/en/5.0/topics/files/).

Note: it's theoretically possible to have the ImageField point to the original file in the source code, but this creates a mess when some corpora don't save their data in the source code. I think it's better to save all data in MEDIA_ROOT.

### Documentation files

Create a model `CorpusDocument` with the following fields:

- `corpus_configuration`: a `ForeignKey` to the CorpusConfiguration.
- `type`: a field with `choices` for the type of document: general information, word models documentation, citation info. Possibly new categories like copyright/licence info (#1010).
- `content`: a `TextField` with the markdown contents.

The contents of these documentation files isn't that big, so I don't think it's an issue to store it directly. In a form, it also makes more sense to edit the contents directly in the interface, rather than download/upload a file.

Using a single `CorpusDocument` model, rather than a separate field for each type of document, also makes the system more flexible for offering different types of documentation.

### Validation

If you're creating a corpus through the interface, it makes sense that the image and documentation may be added at the very end of that process, since they're more the "fluff". The image and documentation should not be required in the database, but having an image should be a requirement for `corpus.ready_to_publish()`.

### PR DESCRIPTION
**Pull request by lukavdplas**
Updates the database model and API for corpus documentation. (close #1524)

The contents of documentation pages are imported into the database, which will allow us to support documentation for database-only corpora.

Instead of implementing each type of documentation separately, there is now a single `CorpusDocumentationPage` model. The API is updated so the frontend requests all documentation pages for a corpus in a single endpoint, rather than using an endpoint for each page.

Other changes:
- Corpora now support two new types of documentation: licence and terms of service (close #1010).
- Templating is now supported on _all_ documentation pages, not just the citation guidelines. (Not really necessary, it was just simpler to make this generally available than to write an exception for citations.)
- Documentation pages can be viewed and edited in the admin (though like the rest of the corpus configuration, changes are reset when the corresponding Python corpus definition is imported).
- The default expectations for the corpus directory structure (e.g. general documentation should be in a folder named "description") are still the same, but you can now override that per corpus.
- Fixes the never-ending loading spinner described in #1256.

### CODE REVIEW

File: backend/addcorpus/models.py [hunk 1]
```diff
@@ -408,3 +397,35 @@ def clean(self):
                     'the corpus or correct the following errors.',
                     e
                 ])
+
+class CorpusDocumentationPage(models.Model):
+    class PageType(models.TextChoices):
```
Comment (JeltevanBoheemen) [inline backend/addcorpus/models.py:402]: I didn't know this base class existed for choices. Much easier to read then the nested tuple business.

File: backend/addcorpus/models.py [hunk 2]
```diff
@@ -408,3 +397,35 @@ def clean(self):
...
+    content = models.TextField(
+        help_text='markdown contents of the documentation'
+    )
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=['corpus_configuration', 'type'],
+                name='unique_documentation_type_for_corpus'
+            )
```
Comment (JeltevanBoheemen) [inline backend/addcorpus/models.py:430]: clean.

File: documentation/Corpus-documentation.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Corpus documentation pages
+
+Corpora can include markdown templates containing documentation. I-analyzer supports several types of documentation pages.
+
+- general information
+- citation guidelines
+- documentation on word models
+- a licence for re-using the data
+- terms of service (usually pro)
```
Comment (JeltevanBoheemen) [inline documentation/Corpus-documentation.md:9]: Define `usually pro` ?

File: documentation/Corpus-documentation.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+All are optional.
+
+### Writing documentation
+
+Documentation files are written in markdown. For information about supported flavours, see [marked.js documentation](https://marked.js.org/) - the frontend uses that package with default options.
```
Comment (JeltevanBoheemen) [inline documentation/Corpus-documentation.md:15]: Maybe refer to an existing documentation page in the source code for inspiration.

**Review by JeltevanBoheemen — APPROVED** [review]
Really important change for the scientific credibility of the application, nicely done.
I'm 50/50 on supporting different paths to documentation, I would be fine with enforcing the default values. But it's opt-in so it's okay.

File: documentation/Corpus-documentation.md (hunk 3 above)
Comment (lukavdplas) [reply documentation/Corpus-documentation.md:9]: ah, I think I just walked away from the computer halfway through that sentence.... 😅

**Pull request merged**